exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceConfigurationTest.java

```diff
@@ -78,29 +85,40 @@ public void setAllConfigurationFields() {
   }
 
   @Test
-  public void disallowNullProjectId() {
+  public void disallowNullOrEmptyProjectId() {
     TraceConfiguration.Builder builder = TraceConfiguration.builder();
 
-    assertThrows(NullPointerException.class, () -> builder.setProjectId(null));
-  }
-
-  @Test
-  public void disallowEmptyProjectId() {
-    TraceConfiguration.Builder builder = TraceConfiguration.builder();
-
-    builder.setProjectId("");
-
-    assertThrows(IllegalArgumentException.class, builder::build);
+    assertThrows(IllegalArgumentException.class, () -> builder.setProjectId(null));
+    assertThrows(IllegalArgumentException.class, () -> builder.setProjectId(""));
   }
 
   @Test
   public void allowToUseDefaultProjectId() {
-    String defaultProjectId = ServiceOptions.getDefaultProjectId();
-    // some test providers might not have project IDs set up
-    if (defaultProjectId != null) {
-      TraceConfiguration configuration = TraceConfiguration.builder().build();
-
-      assertEquals(defaultProjectId, configuration.getProjectId());
+    // some test providers might not have project IDs set up - so we use mocks
+    try (MockedStatic<ServiceOptions> mockedServiceOptions =
```

Comment:
I don't think you need to create the entire `TraceExporter` in this mock.

instead you should be able to just pull the `ProjectId` provider from the default configuration and check that, like the original test.

That would dramatically simplify this test.

Reply:
The original test had a conditional check if `ServiceOptions.getDefaultProjectId()` would return null, the test was essentially no-op. 

This `== null` condition returned true when run on GitHub Actions environment where GCP metadata does not exist.  
Mocking makes this test run independent of environment. 

I can add `null` condition check and make test simple if this seems too complex, but then this test is not adding too much value when run as a GitHub action.

Reply:
I'm not objecting to Mockito, I'm objecting to how you're getting the call to `getDefaultProjectId` to trigger.

Specifically, you're instantiating the exporter and calling `exporter.export(Collections.emptyList())`.

I think you only have to construct default Configuration and then use the ProjectIdSupplier.

In other words, if this is a unit test, you should try to test only using the TraceConfiguration class and its dependencies.